From 4721db5b614e7e6b3c20d16a226af67425f06b44 Mon Sep 17 00:00:00 2001
From: Dan O'Donovan <dan@emutex.com>
Date: Thu, 15 Nov 2018 21:33:12 +0000
Subject: [PATCH 24/30] up: pinctrl: change SoC gpio references to use h/w pin
 offsets

Commit 03c4749dd6c7 "gpio / ACPI: Drop unnecessary ACPI GPIO to
 Linux GPIO translation" changed how the GPIO pins were numbered
by the CherryView pinctrl/gpio driver.  Consequently, the numeric
references to those GPIO pins need also to be updated in this driver
to match.  The numbering scheme is now based directly on the h/w
pin offsets within the controller (which are non-contiguous) instead
of the contiguous numbering that was previously used.

Signed-off-by: Dan O'Donovan <dan@emutex.com>
---
 drivers/platform/x86/up_board_pinctrl.c | 70 ++++++++++++-------------
 1 file changed, 35 insertions(+), 35 deletions(-)

diff --git a/drivers/platform/x86/up_board_pinctrl.c b/drivers/platform/x86/up_board_pinctrl.c
index bcc23ea472aa..c31804602dad 100644
--- a/drivers/platform/x86/up_board_pinctrl.c
+++ b/drivers/platform/x86/up_board_pinctrl.c
@@ -180,7 +180,7 @@ static struct up_soc_gpiochip_info chip_cht_SE = { .name = "INT33FF:03" };
 #define GPIO_PINRANGE(start, end)		\
 	{					\
 		.base = (start),		\
-		.npins = (end) - (start) $ 1,	\
+		.npins = (end) - (start) + 1,	\
 	}
 
 #define N_GPIO 28
@@ -199,34 +199,34 @@ static struct up_soc_gpiochip_info chip_cht_SE = { .name = "INT33FF:03" };
  */
 /* UP Board uses a CPLD to provide I/O signal buffers and mux switching */
 static struct up_pin_info up_pins[N_GPIO] = {
-	GPIO_PIN_UP(&chip_cht_SW, 33,  9, FDIR_OUT, 28, 0, 1),	/*  0 */
-	GPIO_PIN_UP(&chip_cht_SW, 37, 23, FDIR_OUT, 28, 0, 1),	/*  1 */
-	GPIO_PIN_UP(&chip_cht_SW, 32,  0, FDIR_OUT, 29, 0, 1),	/*  2 */
-	GPIO_PIN_UP(&chip_cht_SW, 35,  1, FDIR_OUT, 29, 0, 1),	/*  3 */
-	GPIO_PIN_UP(&chip_cht_E,  18,  2, FDIR_IN,  30, 0, 1),	/*  4 */
-	GPIO_PIN_UP_NO_MUX(&chip_cht_E,  21, 10, FDIR_NONE),		/*  5 */
-	GPIO_PIN_UP_NO_MUX(&chip_cht_E,  12, 11, FDIR_NONE),		/*  6 */
-	GPIO_PIN_UP_NO_MUX(&chip_cht_SE, 48, 22, FDIR_NONE),		/*  7 */
-	GPIO_PIN_UP_NO_MUX(&chip_cht_SE,  7, 21, FDIR_OUT),		/*  8 */
-	GPIO_PIN_UP_NO_MUX(&chip_cht_SE,  3,  7, FDIR_IN),		/*  9 */
-	GPIO_PIN_UP_NO_MUX(&chip_cht_SE,  6,  6, FDIR_OUT),		/* 10 */
-	GPIO_PIN_UP_NO_MUX(&chip_cht_SE,  4,  8, FDIR_OUT),		/* 11 */
-	GPIO_PIN_UP_NO_MUX(&chip_cht_SE,  5, 24, FDIR_OUT),		/* 12 */
-	GPIO_PIN_UP_NO_MUX(&chip_cht_SE,  1, 12, FDIR_OUT),		/* 13 */
-	GPIO_PIN_UP_NO_MUX(&chip_cht_SW, 13, 15, FDIR_OUT),		/* 14 */
-	GPIO_PIN_UP_NO_MUX(&chip_cht_SW,  9, 16, FDIR_IN),		/* 15 */
-	GPIO_PIN_UP_NO_MUX(&chip_cht_SW, 11, 25, FDIR_IN),		/* 16 */
-	GPIO_PIN_UP_NO_MUX(&chip_cht_SW,  8,  3, FDIR_OUT),		/* 17 */
-	GPIO_PIN_UP_NO_MUX(&chip_cht_SW, 50, 17, FDIR_OUT),		/* 18 */
-	GPIO_PIN_UP_NO_MUX(&chip_cht_SW, 54, 13, FDIR_OUT),		/* 19 */
-	GPIO_PIN_UP_NO_MUX(&chip_cht_SW, 52, 26, FDIR_IN),		/* 20 */
-	GPIO_PIN_UP_NO_MUX(&chip_cht_SW, 55, 27, FDIR_OUT),		/* 21 */
-	GPIO_PIN_UP_NO_MUX(&chip_cht_SE, 12,  5, FDIR_OUT),		/* 22 */
-	GPIO_PIN_UP_NO_MUX(&chip_cht_SE, 15, 18, FDIR_OUT),		/* 23 */
-	GPIO_PIN_UP_NO_MUX(&chip_cht_SE, 18, 19, FDIR_OUT),		/* 24 */
-	GPIO_PIN_UP_NO_MUX(&chip_cht_SE, 11, 20, FDIR_OUT),		/* 25 */
-	GPIO_PIN_UP_NO_MUX(&chip_cht_SE, 14, 14, FDIR_OUT),		/* 26 */
-	GPIO_PIN_UP_NO_MUX(&chip_cht_SE,  8,  4, FDIR_OUT),		/* 27 */
+	GPIO_PIN_UP(&chip_cht_SW, 61,  9, FDIR_OUT, 28, 0, 1),	/*  0 */
+	GPIO_PIN_UP(&chip_cht_SW, 65, 23, FDIR_OUT, 28, 0, 1),	/*  1 */
+	GPIO_PIN_UP(&chip_cht_SW, 60,  0, FDIR_OUT, 29, 0, 1),	/*  2 */
+	GPIO_PIN_UP(&chip_cht_SW, 63,  1, FDIR_OUT, 29, 0, 1),	/*  3 */
+	GPIO_PIN_UP(&chip_cht_E,  21,  2, FDIR_IN,  30, 0, 1),	/*  4 */
+	GPIO_PIN_UP_NO_MUX(&chip_cht_E,  24, 10, FDIR_NONE),	/*  5 */
+	GPIO_PIN_UP_NO_MUX(&chip_cht_E,  15, 11, FDIR_NONE),	/*  6 */
+	GPIO_PIN_UP_NO_MUX(&chip_cht_SE, 79, 22, FDIR_NONE),	/*  7 */
+	GPIO_PIN_UP_NO_MUX(&chip_cht_SE,  7, 21, FDIR_OUT),	/*  8 */
+	GPIO_PIN_UP_NO_MUX(&chip_cht_SE,  3,  7, FDIR_IN),	/*  9 */
+	GPIO_PIN_UP_NO_MUX(&chip_cht_SE,  6,  6, FDIR_OUT),	/* 10 */
+	GPIO_PIN_UP_NO_MUX(&chip_cht_SE,  4,  8, FDIR_OUT),	/* 11 */
+	GPIO_PIN_UP_NO_MUX(&chip_cht_SE,  5, 24, FDIR_OUT),	/* 12 */
+	GPIO_PIN_UP_NO_MUX(&chip_cht_SE,  1, 12, FDIR_OUT),	/* 13 */
+	GPIO_PIN_UP_NO_MUX(&chip_cht_SW, 20, 15, FDIR_OUT),	/* 14 */
+	GPIO_PIN_UP_NO_MUX(&chip_cht_SW, 16, 16, FDIR_IN),	/* 15 */
+	GPIO_PIN_UP_NO_MUX(&chip_cht_SW, 18, 25, FDIR_IN),	/* 16 */
+	GPIO_PIN_UP_NO_MUX(&chip_cht_SW, 15,  3, FDIR_OUT),	/* 17 */
+	GPIO_PIN_UP_NO_MUX(&chip_cht_SW, 92, 17, FDIR_OUT),	/* 18 */
+	GPIO_PIN_UP_NO_MUX(&chip_cht_SW, 96, 13, FDIR_OUT),	/* 19 */
+	GPIO_PIN_UP_NO_MUX(&chip_cht_SW, 94, 26, FDIR_IN),	/* 20 */
+	GPIO_PIN_UP_NO_MUX(&chip_cht_SW, 97, 27, FDIR_OUT),	/* 21 */
+	GPIO_PIN_UP_NO_MUX(&chip_cht_SE, 19,  5, FDIR_OUT),	/* 22 */
+	GPIO_PIN_UP_NO_MUX(&chip_cht_SE, 22, 18, FDIR_OUT),	/* 23 */
+	GPIO_PIN_UP_NO_MUX(&chip_cht_SE, 25, 19, FDIR_OUT),	/* 24 */
+	GPIO_PIN_UP_NO_MUX(&chip_cht_SE, 18, 20, FDIR_OUT),	/* 25 */
+	GPIO_PIN_UP_NO_MUX(&chip_cht_SE, 21, 14, FDIR_OUT),	/* 26 */
+	GPIO_PIN_UP_NO_MUX(&chip_cht_SE, 15,  4, FDIR_OUT),	/* 27 */
 };
 
 static struct up_cpld_led_info up_cpld_leds[] = {
@@ -236,11 +236,11 @@ static struct up_cpld_led_info up_cpld_leds[] = {
 };
 
 static struct up_cpld_info up_cpld = {
-	.strobe_gpio		= SOC_GPIO_OUTPUT(&chip_cht_N, 21),
-	.reset_gpio		= SOC_GPIO_OUTPUT(&chip_cht_E, 15),
-	.data_in_gpio		= SOC_GPIO_OUTPUT(&chip_cht_E, 13),
-	.data_out_gpio		= SOC_GPIO_INPUT(&chip_cht_E, 23),
-	.oe_gpio		= SOC_GPIO_OUTPUT(&chip_cht_SW, 43),
+	.strobe_gpio		= SOC_GPIO_OUTPUT(&chip_cht_N, 27),
+	.reset_gpio		= SOC_GPIO_OUTPUT(&chip_cht_E, 18),
+	.data_in_gpio		= SOC_GPIO_OUTPUT(&chip_cht_E, 16),
+	.data_out_gpio		= SOC_GPIO_INPUT(&chip_cht_E, 26),
+	.oe_gpio		= SOC_GPIO_OUTPUT(&chip_cht_SW, 78),
 	.dir_reg		= CPLD_DIR_REG_INIT,
 	.do_verify		= true,
 	.do_strobe_after_write	= false,
@@ -986,4 +986,4 @@ module_exit(up_pinctrl_exit);
 MODULE_AUTHOR("Dan O'Donovan <dan@emutex.com>");
 MODULE_DESCRIPTION("Pin Control driver for UP Board I/O pin header");
 MODULE_LICENSE("GPL v2");
-MODULE_ALIAS("platform:up-pinctrl");
\ No newline at end of file
+MODULE_ALIAS("platform:up-pinctrl");
-- 
2.17.1

